src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -1172,17 +1166,36 @@ public void StartInfo_NotepadWithContent_withArgumentList(bool useShellExecute)
 
                 try
                 {
-                    process.WaitForInputIdle(); // Give the file a chance to load
-                    Assert.Equal("notepad", process.ProcessName);
-
-                    // On some Windows versions, the file extension is not included in the title
-                    Assert.StartsWith(Path.GetFileNameWithoutExtension(tempFile), process.MainWindowTitle);
+                    VerifyNotepadMainWindowTitle(process, tempFile);
                 }
                 finally
                 {
                     process?.Kill();
                 }
             }
         }
+
+        private void VerifyNotepadMainWindowTitle(Process process, string filename)
+        {
+            // On some Windows versions, the file extension is not included in the title
+            string expected = Path.GetFileNameWithoutExtension(filename);
+
+            process.WaitForInputIdle(); // Give the file a chance to load
+            Assert.Equal("notepad", process.ProcessName);
+
+            // Notepad calls CreateWindowEx with pWindowName of empty string, then calls SetWindowTextW
+            // with "Untitled - Notepad" then finally if you're opening a file, calls SetWindowTextW
+            // with something similar to "myfilename - Notepad". So there's a race between input idle
+            // and the expected MainWindowTitle because of how Notepad is implemented.
+            Thread.Sleep(500);
+            string title = process.MainWindowTitle;
+            if (!title.Contains(expected))
+            {
+                Thread.Sleep(5000);
+                title = process.MainWindowTitle;
```

Comment:
Does this need to call `Refresh()`?

`MainWindowTitle` is cached. I believe that this will keep returning the stale value once you get into the loop.